lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -669,7 +669,7 @@ func sufficientHints(nHintsLeft int, currentAmount,
 		return true
 	}
 
-	if currentAmount >= targetAmount {
+	if targetAmount != 0 && currentAmount >= targetAmount {
```

Comment:
Can we also update our itest to cover this case?

Reply:
actually was referring to checking the `targetAmount != 0` part, and see you already have it added here, nice!